### PR TITLE
EIP-1559 - Improve gas timing logic to show more accurate verbiage

### PR DESCRIPTION
**Pull request by darkwing**
Updates to the logic to make the verbiage make more sense:

https://consensys.slack.com/archives/C021UN08FBJ/p1627504435042300

https://user-images.githubusercontent.com/46655/127565008-a644063c-29f9-46e5-ac5c-a43f20aa1bfd.mp4

### CODE REVIEW

File: ui/components/app/gas-timing/gas-timing.component.js [hunk 1]
```diff
@@ -8,28 +8,65 @@ import { useGasFeeEstimates } from '../../../hooks/useGasFeeEstimates';
...
-    const seconds = Math.ceil(milliseconds / 1000);
-    if (seconds <= SECOND_CUTOFF) {
-      return t('gasTimingSeconds', [seconds]);
+  // If the user has chosen a value lower than the low gas fee estimate,
+  // We'll need to use the useEffect hook below to make a call to calculate
+  // the time to show
+  const isUnknownLow =
+    gasFeeEstimates?.low &&
+    Number(maxPriorityFeePerGas) <
+      Number(gasFeeEstimates.low.suggestedMaxPriorityFeePerGas);
```
Comment (danjm) [inline ui/components/app/gas-timing/gas-timing.component.js:48]: Do optional chaining incase `gasFeeEstimates.low` is undefined

File: ui/components/app/gas-timing/gas-timing.component.js [hunk 2]
```diff
@@ -42,36 +79,66 @@ export default function GasTiming({ maxPriorityFeePerGas }) {
   const { low, medium, high } = gasFeeEstimates;
```
Comment (danjm) [inline ui/components/app/gas-timing/gas-timing.component.js:79]: Can we default these low medium and highs to empty objects

File: ui/components/app/gas-timing/gas-timing.component.js [hunk 3]
```diff
@@ -8,28 +8,65 @@ import { useGasFeeEstimates } from '../../../hooks/useGasFeeEstimates';
...
+  const isUnknownLow =
+    gasFeeEstimates?.low &&
+    Number(maxPriorityFeePerGas) <
+      Number(gasFeeEstimates.low.suggestedMaxPriorityFeePerGas);
+
+  useEffect(() => {
+    const priority = maxPriorityFeePerGas;
+    const fee = maxFeePerGas;
+
+    if (isUnknownLow) {
```
Comment (danjm) [inline ui/components/app/gas-timing/gas-timing.component.js:54]: We probably also need to call `getGasTimeEstimate` in situations where `isUnkownLow` is false. For instance, if `maxPriorityFeePerGas` is greater than `gasFeeEstimates.low.suggestedMaxPriorityFeePerGas`, and then either `maxPriorityFeePerGas` or `maxFeePerGas` change

**Review by danjm — APPROVED** [review]

**Comment by danjm** [comment]
LGTM!

File: ui/components/app/gas-timing/gas-timing.component.js [hunk 4]
```diff
@@ -83,4 +164,5 @@ export default function GasTiming({ maxPriorityFeePerGas }) {
 
 GasTiming.propTypes = {
   maxPriorityFeePerGas: PropTypes.string.isRequired,
+  maxFeePerGas: PropTypes.string.isRequired,
```
Comment (dan437) [inline ui/components/app/gas-timing/gas-timing.component.js:167]: Note for myself: I will need to pass `maxFeePerGas` when calling the `GasTiming` component from Swaps: https://github.com/MetaMask/metamask-extension/blob/develop/ui/pages/swaps/fee-card/fee-card.js#L144

I will send a small PR for that once this one is merged.

**Review by dan437 — APPROVED** [review]

File: ui/components/app/gas-timing/gas-timing.component.js (hunk 4 above)
Comment (dan437) [reply ui/components/app/gas-timing/gas-timing.component.js:167]: Here is a PR for that: https://github.com/MetaMask/metamask-extension/pull/11684

**Pull request merged**